core/src/main/scala/io/finch/request/RequestReader.scala

```diff
@@ -205,4 +205,36 @@ object RequestReader {
       val item = i
       def apply(req: R): Future[A] = f(req)
     }
+
+  import scala.reflect.ClassTag
+  import shapeless._, labelled.{FieldType, field}
+
+  class GenericDerivation[A] {
+    def fromParams[Repr <: HList](implicit
+      gen: LabelledGeneric.Aux[A, Repr],
+      fp: FromParams[Repr]
+    ): RequestReader[A] = fp.reader.map(gen.from)
+  }
+
+  trait FromParams[L <: HList] {
+    def reader: RequestReader[L]
+  }
+
+  object FromParams {
```

Comment:
I like it! Do you think we should do the same trick for headers (is it useful?) and bodies. Like:

```scala
val a = RequestReader.for[User].fromParams
val b = RequestReader.for[User].fromHeaders
val c = RequestReader.for[User].fromBody
```